modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -6662,6 +6662,8 @@ public List<CodegenParameter> fromRequestBodyToFormParameters(RequestBody body,
                 // Set 'required' flag defined in the schema element
                 if (!codegenParameter.required && schema.getRequired() != null) {
                     codegenParameter.required = schema.getRequired().contains(entry.getKey());
+                } else if (!codegenParameter.required) {
+                    codegenParameter.required = allRequired.stream().anyMatch(r -> r.equals(codegenParameter.paramName));
```

Comment:
can you please add a 1-liner comment stating this is for allOf schema?

Reply:
done